bazel/cc_toolchains/clang_cc_toolchain_config.bzl

```diff
@@ -511,8 +511,14 @@ def _impl(ctx):
 
     # With clang 14 and lower, we expect it to be built with libc++ debug
     # support. In later LLVM versions, we expect the assertions define to work.
+    # clang 17 deprecates LIBCPP_ENABLE_ASSERTIONS in favor of HARDENED_MODE.
+    # TODO(samkhn): Once LLVM makes ENABLE_ASSERTIONS equivalent to
+    # ENABLE_HARDENING, remove the HARDENED_MODE check
+    # https://github.com/llvm/llvm-project/blob/main/libcxx/include/__config
```

Comment:
Reading https://reviews.llvm.org/D154997, I think the intent is to replace LIBCPP_ENABLE_ASSERTIONS with LIBCPP_ENABLE_HARDENED_MODE. So the change in v18 would be to drop support for LIBCPP_ENABLE_ASSERTIONS (or possibly make it an error). Wouldn't we then still need to keep setting LIBCPP_ENABLE_HARDENED_MODE to get the current behavior?

Reply:
Ah thanks for catching that. Removed this with 4c685f4